lib/node_modules/@stdlib/ndarray/some-by/lib/assign.js

```diff
@@ -0,0 +1,205 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isFunction = require( '@stdlib/assert/is-function' );
+var isObject = require( '@stdlib/assert/is-object' );
+var isInteger = require( '@stdlib/assert/is-integer' );
+var contains = require( '@stdlib/assert/contains' );
+var isndarrayLike = require( '@stdlib/assert/is-ndarray-like' );
+var unaryReduceSubarrayBy = require( '@stdlib/ndarray/base/unary-reduce-subarray-by' );
+var ndims = require( '@stdlib/ndarray/ndims' );
+var base = require( '@stdlib/ndarray/base/some-by' );
+var getDtype = require( '@stdlib/ndarray/dtype' );
+var getShape = require( '@stdlib/ndarray/shape' ); // note: non-base accessor is intentional due to the input arrays originating in userland
+var getOrder = require( '@stdlib/ndarray/base/order' );
+var maybeBroadcastArray = require( '@stdlib/ndarray/base/maybe-broadcast-array' );
+var broadcastScalar = require( '@stdlib/ndarray/base/broadcast-scalar' );
+var objectAssign = require( '@stdlib/object/assign' );
+var zeroTo = require( '@stdlib/array/base/zero-to' );
+var format = require( '@stdlib/string/format' );
+var defaults = require( './defaults.json' );
+var validate = require( './validate.js' );
+
+
+// VARIABLES //
+
+var TYPES = [
+	'int8',
+	'int16',
+	'int32',
+	'uint8',
+	'uint8c',
+	'uint16',
+	'uint32'
+];
+
+
+// MAIN //
+
+/**
+* Tests whether at least `n` elements in an ndarray pass a test implemented by a predicate function along one or more dimensions.
+*
+* @param {ndarray} x - input ndarray
+* @param {(ndarray|*)} n - number of elements which must pass the test
+* @param {ndarray} y - output ndarray
+* @param {Options} [options] - function options
+* @param {IntegerArray} [options.dims] - list of dimensions over which to perform a reduction
+* @param {Function} predicate - predicate function
+* @param {*} [thisArg] - predicate execution context
+* @throws {TypeError} first argument must be an ndarray-like object
+* @throws {Error} second argument must be broadcast-compatible with the non-reduced dimensions of the input ndarray
+* @throws {TypeError} third argument must be an ndarray-like object
+* @throws {TypeError} options argument must be an object
+* @throws {Error} must provide valid options
+* @throws {TypeError} predicate must be a function
```

Comment:
```suggestion
* @throws {TypeError} callback argument must be a function
```

Use similar description as in `count-if`.